### PR TITLE
[IMP] vuestorefront: invoices search

### PR DESCRIPTION
**Pull request by oerp-odoo**
It is now possible to search current partner invoices by Ids and invoice line names (ilike).

### CODE REVIEW

File: vuestorefront/models/account_move.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    _inherit = 'account.move'
+
+    @api.model
+    def prepare_vsf_domain(self, commercial_partner_id, **kwargs):
+        domain = [('message_partner_ids', 'child_of', [commercial_partner_id])]
+        if kwargs.get('ids'):
+            domain.append(('id', 'in', kwargs['ids']))
+        if kwargs.get('line_name'):
+            domain.append(('invoice_line_ids.name', 'ilike', kwargs['line_name']))
+        return domain
```
Comment (DarkoNikolovski) [inline vuestorefront/models/account_move.py:24]: "Line name" should be just addition to the search options that are currently available.

**Review by DarkoNikolovski — COMMENTED** [review]
I can't see how it will be possible at the moment to get info about the order related to the invoice, because some of the information related to the order are shown on the invoice row as well.

Also some of the search options from the existing search in the portal are missing.

**Comment by oerp-odoo** [comment]
> I can't see how it will be possible at the moment to get info about the order related to the invoice, because some of the information related to the order are shown on the invoice row as well.
> 
> Also some of the search options from the existing search in the portal are missing.

What do you mean some options are missing? I thought this task was to implement single option to search by invoice line name. What else should be done then?

File: vuestorefront/models/account_move.py (hunk 1 above)
Comment (oerp-odoo) [reply vuestorefront/models/account_move.py:24]: Originally there were 0 search options. What kind of options you have in mind here?

Comment (DarkoNikolovski) [reply vuestorefront/models/account_move.py:24]: @Ernestas-t maybe you can clarify the requirements?

**Review by DarkoNikolovski — APPROVED** [review]

File: vuestorefront/schemas/objects.py [hunk 2]
```diff
@@ -27,6 +27,15 @@
                                                 ('ToInvoice', 'to invoice'), ('NothingtoInvoice', 'no')])
 
 InvoiceState = graphene.Enum('InvoiceState', [('Draft', 'draft'), ('Posted', 'posted'), ('Cancelled', 'cancel')])
+InvoicePaymentState = graphene.Enum(
+    'InvoicePaymentState',
+    [
+        ('NotPaid', 'not_paid'),
+        ('Paid', 'paid'),
+        ('PartiallyPaid', 'partial'),
```
Comment (Drsmartinkus) [inline vuestorefront/schemas/objects.py:35]:
```suggestion
        ('Partially Paid', 'partial'),
```

File: vuestorefront/schemas/objects.py [hunk 3]
```diff
@@ -27,6 +27,15 @@
                                                 ('ToInvoice', 'to invoice'), ('NothingtoInvoice', 'no')])
 
 InvoiceState = graphene.Enum('InvoiceState', [('Draft', 'draft'), ('Posted', 'posted'), ('Cancelled', 'cancel')])
+InvoicePaymentState = graphene.Enum(
+    'InvoicePaymentState',
+    [
+        ('NotPaid', 'not_paid'),
```
Comment (Drsmartinkus) [inline vuestorefront/schemas/objects.py:33]:
```suggestion
        ('Not Paid', 'not_paid'),
```

**Review by Drsmartinkus — APPROVED** [review]

File: vuestorefront/schemas/objects.py (hunk 2 above)
Comment (oerp-odoo) [reply vuestorefront/schemas/objects.py:35]: No, we can't use `space` here. VSF will fail. It expects value specifically as CamelCase (note, this is not odoo selection field).

**Pull request merged**